osgp/platform/osgp-adapter-kafka-distributionautomation/src/main/java/org/opensmartgridplatform/adapter/kafka/da/application/services/DomainHelperService.java

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright 2020 Smart Society Services B.V.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except in compliance with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ */
+package org.opensmartgridplatform.adapter.kafka.da.application.services;
+
+import org.opensmartgridplatform.domain.core.entities.Device;
+import org.opensmartgridplatform.domain.core.entities.Organisation;
+import org.opensmartgridplatform.domain.core.entities.RtuDevice;
+import org.opensmartgridplatform.domain.core.exceptions.NotAuthorizedException;
+import org.opensmartgridplatform.domain.core.exceptions.UnknownEntityException;
+import org.opensmartgridplatform.domain.core.repositories.OrganisationRepository;
+import org.opensmartgridplatform.domain.core.repositories.RtuDeviceRepository;
+import org.opensmartgridplatform.domain.core.services.SecurityService;
+import org.opensmartgridplatform.domain.core.valueobjects.DeviceFunction;
+import org.opensmartgridplatform.shared.exceptionhandling.ComponentType;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalException;
+import org.opensmartgridplatform.shared.exceptionhandling.FunctionalExceptionType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Service;
+
+@Service
+public class DomainHelperService {
+
+    private static final ComponentType COMPONENT_TYPE = ComponentType.KAFKA_DISTRIBUTION_AUTOMATION;
+
+    @Autowired
+    private RtuDeviceRepository rtuDeviceRepository;
+
+    @Autowired
+    private OrganisationRepository organisationRepository;
+
+    @Autowired
+    private SecurityService securityService;
+
+    public RtuDevice findDevice(final String deviceIdentification) throws FunctionalException {
+        final RtuDevice device = this.rtuDeviceRepository.findByDeviceIdentification(deviceIdentification);
+        if (device == null) {
```

Comment:
Why don't we use an optional with orElseThrow?

Reply:
Not all repositories are already updated to return optionals. Dependent of the usage of the methods throughout the code, refactoring to use optionals might be a lot of work.
In this case the method in the repository was referenced only a few times, so therefor I
refactored it to use an optional. 
The findOrganisation method in this class could also use the same refactoring, however updating the organisationRepository will have a greater impact (more references in more value streams), so I'll guess I won't do this now...